pkg/db/db.go

```diff
@@ -628,13 +628,16 @@ func (db *DB) GetLeases(input GetLeasesInput) (GetLeasesOutput, error) {
 	filters := make([]string, 0)
 	filterValues := make(map[string]*dynamodb.AttributeValue)
 
+	queryInput := &dynamodb.QueryInput{
+		TableName:      aws.String(db.LeaseTableName),
+		ConsistentRead: aws.Bool(db.ConsistentRead),
```

Comment:
Would this work given that no index is given?

I think the easist way to accomplish this is
1. Go back to largely how it was
2. Make a variable called, queryable, and queryIndex (or something like that) and mark it false to start
3. If one of the index filters are set, set the field (make sure not to overwrite, we probably want to prioritize principal_id)
4. Make the query with filters if queryable, make the scan with filters otherwise
5. Output the results to the same spot